ludwig/models/modules/category_encoders.py

```diff
@@ -0,0 +1,117 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import logging
+
+import tensorflow as tf
+from tensorflow.keras.layers import Layer
+from ludwig.models.modules.embedding_modules import Embed
+
+logger = logging.getLogger(__name__)
+
+
+class CategoricalPassthroughEncoder(Layer):
+
+    def __init__(
+            self,
+            **kwargs
+    ):
+        super(CategoricalPassthroughEncoder, self).__init__()
+
+    def call(self, inputs, training=None, mask=None):
+        """
+            :param inputs: The inputs fed into the encoder.
+                   Shape: [batch x 1], type tf.int32
+        """
+        return inputs
+
+
+class CategoricalEmbedEncoder(Layer):
+
+    def __init__(
+            self,
+            vocab=None,
+            embedding_size=None,
+            representation=None,
+            embeddings_trainable=None,
+            pretrained_embeddings=None,
+            embeddings_on_cpu=None,
+            dropout=None,
+            initializer=None,
+            regularize=None,
+            **kwargs
+    ):
+        super(CategoricalEmbedEncoder, self).__init__()
+
+        self.vocab = vocab
+        self.embedding_size = embedding_size
+        self.representation = representation
+        self.embeddings_trainable = embeddings_trainable
+        self.pretrained_embeddings = pretrained_embeddings
+        self.embeddings_on_cpu = embeddings_on_cpu
+        self.dropout = dropout
+        self.initializer = initializer
+        self.regularize = regularize
+
+
+        self.embed = Embed(
+            vocab=self.vocab,
+            embedding_size=self.embedding_size,
+            representation=self.representation,
+            embeddings_trainable=self.embeddings_trainable,
+            pretrained_embeddings=self.pretrained_embeddings,
+            embeddings_on_cpu=self.embeddings_on_cpu,
+            dropout=self.dropout,
+            initializer=self.initializer,
+            regularize=self.regularize
+        )
+
+        pass
+
+    def call(self, inputs, training=None, mask=None):
+        """
+            :param inputs: The inputs fed into the encoder.
+                   Shape: [batch x 1], type tf.int32
+
+            :param return: embeddings of shape [batch x embed size], type tf.float32
+        """
+
+        embedded, embedding_size = self.embed(
+            tf.cast(inputs, dtype=tf.int32),
+            None,   # testing...need regularizer
+            self.dropout,
+            self.embeddings_trainable
+        )
+
+        embedded = tf.squeeze(embedded)
```

Comment:
why is this needed?

Reply:
the shape created by `self.embed()` function is (batch, 1, embed_size).  Without the `tf.squeeze()`, the `tf.concat()` fails ValueError because of incompatible dimensions.  From what I can tell, we need a tensor with rank 2 at this point.  So the `tf.squeeze()` changed shape to (batch, embed_size).

Reply:
got it. So we should modify Embed to return the right thing already, but in the meantime we can keep this squeeze. Please add a todo comment here to remove squeeze when it won't be needed anymore

Reply:
added todo